File: card-core/src/main/java/com/jean202/cardmizer/core/port/in/RecommendCardUseCase.java
package com.jean202.cardmizer.core.port.in;

import com.jean202.cardmizer.core.domain.RecommendationContext;
import com.jean202.cardmizer.core.domain.RecommendationResult;

public interface RecommendCardUseCase {
    RecommendationResult recommend(RecommendationContext context);
}

File: card-core/src/main/kotlin/com/jean202/cardmizer/core/port/in/RecommendCardUseCase.kt
package com.jean202.cardmizer.core.port.`in`

import com.jean202.cardmizer.core.domain.RecommendationContext
import com.jean202.cardmizer.core.domain.RecommendationResult

fun interface RecommendCardUseCase {
    fun recommend(context: RecommendationContext): RecommendationResult
}
